### ISSUE TITLE
Importer hanging in Bot Garden > Propagations preprocessing setp

### ISSUE THREAD

**Issue by meganforbes** [collectionspace/collectionspace-csv-importer#75]
I haven't tried other procedures so not sure if the CSV is the issue, my connection, etc. Will do more investigating just wanted to get this sketched out.

Tried two Bot Garden Propagation imports, both hanging in preprocessing. CSV files attached.

[Propagation Sample Data for QA.csv](https://github.com/collectionspace/collectionspace-csv-importer/files/6688518/Propagation.Sample.Data.for.QA.csv)

[5f5f1c69-fe4a-44b4-95c3-4999653e1611.csv](https://github.com/collectionspace/collectionspace-csv-importer/files/6688517/5f5f1c69-fe4a-44b4-95c3-4999653e1611.csv)

**Comment by kspurgin** [comment]
I have no idea how to access the live logs or anything on Heroku, so you can cancel that task and I'll try to reproduce on my own version and see if I can tell what is going on.

**Comment by kspurgin** [comment]
Pretty sure if it was your connection, it wouldn't work at all or you would have gotten some pretty failure like this when creating the connection: #76

**Comment by kspurgin** [comment]
So I found the issue, which appears to be with the JSON mapper/collectionspace-mapper: 

```
[16, 25] in /Users/kristina/code/cspace-batch-import/app/jobs/preprocess_job.rb
   16:     puts "Preprocess StepManagerService started"
   17:
   18:     begin
   19:       begin
   20:         byebug
=> 21:         handler = preprocess.batch.handler
   22:       rescue CollectionSpace::Mapper::IdFieldNotInMapperError
   23:         manager.add_error!
   24:         manager.add_message('The import tool cannot determine the unique ID field for this record type.
   25: Contact import tool admin and ask them to fix the RecordMapper.')
(byebug) preprocess.batch
#<Batch id: 1, name: "prop2", job_id: "2dd8b6cb05e10e798824feca", num_rows: 2, step_state: "preprocessing", status_state: "running", user_id: 1, group_id: 1, connection_id: 5, mapper_id: 206, created_at: "2021-06-21 20:37:11", updated_at: "2021-06-21 20:58:17", batch_config: {}>
(byebug) preprocess.batch.handler
*** KeyError Exception: key not found: "propagations"

nil
```

**Comment by kspurgin** [comment]
So, it looks like one of these is the problem:

- `cspace-config-untangler` is setting the mapper's `config/service_path` value improperly to "propagations"
- `collectionspace-client` doesn't have the service type set up properly

```
[68, 77] in /Users/kristina/.rvm/gems/ruby-2.7.3/bundler/gems/collectionspace-client-4759c1a467bd/lib/collectionspace/client/helpers.rb
   68:       options = prepare_query(query, params)
   69:       request 'GET', query.path, options
   70:     end
   71:
   72:     def service(type:, subtype: '')
=> 73:       CollectionSpace::Service.get(type: type, subtype: subtype)
   74:     end
   75:
   76:     private
   77:
(byebug) type
"propagations"
(byebug) subtype
""
(byebug) s

[4, 13] in /Users/kristina/.rvm/gems/ruby-2.7.3/bundler/gems/collectionspace-client-4759c1a467bd/lib/collectionspace/client/service.rb
    4:   # CollectionSpace service
    5:   class Service
    6:     TERM_SUFFIX = 'TermGroupList/0/termDisplayName'
    7:     def self.get(type:, subtype: '')
    8:       {
=>  9:         'acquisitions' => {
   10:           identifier: 'acquisitionReferenceNumber',
   11:           ns_prefix: 'acquisitions',
   12:           path: 'acquisitions',
   13:           term: nil
(byebug) s

[1, 10] in /Users/kristina/.rvm/gems/ruby-2.7.3/gems/better_errors-2.7.1/lib/better_errors/exception_extension.rb
    1: module BetterErrors
    2:   module ExceptionExtension
    3:     prepend_features Exception
    4:
    5:     def set_backtrace(*)
=>  6:       if caller_locations.none? { |loc| loc.path == __FILE__ }
    7:         @__better_errors_bindings_stack = ::Kernel.binding.callers.drop(1)
    8:       end
    9:
   10:       super
```

**Comment by kspurgin** [comment]
Yay, it's not in my code. (mapper or untangler) :-D 

There is no entry for `propagations` here:
https://github.com/collectionspace/collectionspace-client/blob/master/lib/collectionspace/client/service.rb 

There is also no entry for `insurances` or `transports` here, so those are also not going to work. 

What I first need to maybe fix is making this actually fail instead of just hanging forever.

**Comment by kspurgin** [comment]
The linked pull request above will fix this, in the sense that it will make it fail in an informative way, rather than hanging forever. This is what it looks like on my dev machine with the code in the PR: 

<img width="1367" alt="Screen Shot 2021-06-21 at 20 05 09" src="https://user-images.githubusercontent.com/450570/123000252-58da3400-d37d-11eb-9ef4-aa612ac7870e.png">

Once Mark is back and has merged collectionspace/collectionspace-client#11, then I can: 
- update `collectionspace-mapper` to use the new version of `collectionspace-client`
- update `collectionspace-csv-importer` to use the new version of `collectionspace-mapper` and `collectionspace-client`
- make a PR that will fix the overall lack of support in `collectionspace-client` for some record types
